aggregator_core/src/datastore.rs

```diff
@@ -3952,6 +3952,71 @@ impl<C: Clock> Transaction<'_, C> {
         .await?;
         Ok(())
     }
+
+    #[tracing::instrument(skip(self), err)]
+    pub async fn get_global_hpke_keypair(&self) -> Result<Option<HpkeKeypair>, Error> {
+        let stmt = self
+            .prepare_cached(
+                "SELECT config_id, config, private_key FROM task_hpke_keys WHERE task_id IS NULL;",
+            )
+            .await?;
+
+        let hpke_key_rows = self.query(&stmt, &[]).await?;
+        if hpke_key_rows.is_empty() {
+            Ok(None)
+        } else {
+            let config = HpkeConfig::get_decoded(hpke_key_rows[0].get("config"))?;
+
+            let config_id = u8::try_from(hpke_key_rows[0].get::<_, i16>("config_id"))?;
+            let encrypted_private_key: Vec<u8> = hpke_key_rows[0].get("private_key");
+            let private_key = HpkePrivateKey::new(self.crypter.decrypt(
+                "task_hpke_keys",
+                &config_id.to_be_bytes(),
```

Comment:
This AAD subfield is normally the concatenation of `task_id | config_id`. My original iteration had this be `"janus_global_hpke_config" | config_id`, but I wasn't sure if that really had any benefit so I just went with `config_id` for simplicity. Open to feedback on this.

Reply:
This seems fine, as they are domain separated from per-task keys by the length of the AAD.